### ISSUE TITLE
Circle CI openssl breakage on macos

### ISSUE THREAD

**Issue by bagder** [curl/curl#13438]
### I did this

I made [this PR](https://github.com/curl/curl/pull/13437)

The regular x86 macos CI jobs on Circle CI all fail, with unrelated problems. It seems configure cannot find OpenSSL in those builds.

[example build](https://app.circleci.com/pipelines/github/curl/curl/8274/workflows/cbd411bd-ccd5-4a0e-9def-467a210aee5f/jobs/81537)

OpenSSL is installed with homebrew and nothing has been changed in that setup recently. We also do OpenSSL builds on macOS on GitHub actions that work, so it does not seem to be an obvious homebrew related glitch?

### I expected the following

The CI jobs to work fine.

### curl/libcurl version

current git master

### operating system

CI jobs on macOS

**Comment by bagder** [comment]
Probably related, Circle CI will shut down their x86 macos machines on June 28th:

## Subject: macOS Intel support deprecation 

Hello,

On June 28th, 2024 we will end support for our Intel macOS resources. This change impacts all
customers using Intel Mac resources, across all plans.

In November 2020, Apple released the first Macs with an ARM-based M1 chip and announced that the
transition from Intel processors to their own silicon-based processors would take about two years.
Now that we’re well past that two-year timeframe, we believe we can best serve our customers by
focusing on delivering the latest Apple Silicon resources moving forward.

Any customer that has a macOS project running on an Intel resource will need to update their config
to use a new resource class by June 28th, 2024.

**Comment by dfandrich** [comment]
As of commit d7332e3e4 (last January) we shouldn't have any more x86 builds on Cirrus.

**Comment by bagder** [comment]
I presume you mean Circle CI, not Cirrus.

Oh! I now think my mention of `x86` in the description above is wrong and they are all indeed aarch64 builds. I'll update accordingly.

**Comment by bagder** [comment]
[log](https://app.circleci.com/pipelines/github/curl/curl/8289/workflows/12f4c6e7-0c8c-46f6-9054-93d2f755e936/jobs/81814/parallel-runs/0/steps/0-103)
~~~
ld: library 'crypto' not found
~~~

It looks like maybe it gets the wrong path from pkg-config?

~~~
configure: pkg-config: SSL_LDFLAGS: "-L/opt/homebrew/Cellar/openssl@3/3.3.0"
~~~

**Comment by vszakats** [comment]
Yes, libs are under `-L<brew-prefix>/Cellar/openssl@3/3.3.0/lib` on my system.

```
$ pkg-config --libs openssl
-L/usr/local/Cellar/openssl@3/3.3.0/lib -lssl -lcrypto
$ which pkg-config
/usr/local/bin/pkg-config
```

I wonder where the wrong value is coming from.

**Comment by vszakats** [comment]
Also could not replicate this locally.

```
../configure --enable-warnings --enable-websockets --with-openssl --enable-debug PKG_CONFIG_PATH="$(brew --prefix openssl)/lib/pkgconfig" CPPFLAGS="-I$(brew --prefix libpsl)/include" CFLAGS='-Wno-vla -mmacosx-version-min=10.9' LDFLAGS="-L$(brew --prefix libpsl)/lib -L$(brew --prefix icu4c)/lib" LIBS="-licuuc -licudata" || tail -1000 config.log
[...]
checking for pkg-config... (cached) /usr/local/bin/pkg-config
checking for openssl options with pkg-config... found
configure: pkg-config: SSL_LIBS: "-lssl -lcrypto"
configure: pkg-config: SSL_LDFLAGS: "-L/usr/local/Cellar/openssl@3/3.3.0/lib"
configure: pkg-config: SSL_CPPFLAGS: "-I/usr/local/Cellar/openssl@3/3.3.0/include"
```

**Comment by bagder** [comment]
Also, the macos jobs on GHA does not show this problem.

**Comment by vszakats** [comment]
I think it was this Homebrew issue:
https://github.com/Homebrew/homebrew-core/commit/ff65c06de04cdfd739018aa84a1a86812a1d7a68

With the payload updates just hours ago:
https://github.com/Homebrew/homebrew-core/commit/67b2f1bc91c7d9bfe26b56a284a6ac329e83ece2

**Comment by vszakats** [comment]
It's working fine again now:
https://app.circleci.com/pipelines/github/curl/curl/8292
